app/routes/questionnaire.py

```diff
@@ -187,7 +187,14 @@ def block(schema, questionnaire_store, block_id, list_name=None, list_item_id=No
 
     if "action[clear_radios]" in request.form:
         block_handler.clear_radio_answers()
-        return redirect(request.url)
+        return redirect(
+            url_for(
+                "questionnaire.block",
+                block_id=block_id,
+                list_name=list_name,
+                list_item_id=list_item_id,
+            )
+        )
```

Comment:
We could use the location object itself.
```suggestion
        return redirect(block_handler.current_location.url())
```